lib/Storage/Storage.php

```diff
@@ -195,18 +196,28 @@ public function stat($path) {
 		}
 
 		$size = $file->getProperty(self::SP_PROPERTY_SIZE) ?: FileInfo::SPACE_UNKNOWN;
-		$mtimeValue = $file->getProperty(self::SP_PROPERTY_MTIME);
-		$mtime = $mtimeValue ? new \DateTime($mtimeValue) : null;
+		$mtimeValue = (string)$file->getProperty(self::SP_PROPERTY_MTIME);
+		$name = (string)$file->getProperty(self::SP_PROPERTY_NAME);
+
+		$mtime = new \DateTime($mtimeValue);
+		if($mtimeValue === '') {
+			// SP2013 does not provide an mtime. This way we cause a sync every
+			// 5minutes… hopefully not too often, hopefully not to rarely
+			$i = ((int)date('i') % 5) * 5;
+			$mtime = new \DateTime(date('Y-m-d H:' . $i));
```

Comment:
@icewind1991 or makes returning false respectively time() more sense, when the API does not report back the mtime? Does it make a difference at all, since at least on my dev instance i see scan requests either way.

Reply:
I guess just the timestamp is more sane as you don't have DST mess but @icewind1991 should know

Reply:
Meanhwhile I also think it is better to go with mtime(), as also it is default behaviour when nothing (well, false) is returned. Plus disliking hacks like this.